articles/Frontend/Components/calculator.md

```diff
@@ -0,0 +1,843 @@
+# Calculator component overview
+
+In order to properly support the [mathematical order of operations](https://en.wikipedia.org/wiki/BODMAS)
+we designed another calculator component, able to work on expressions instead of
+applying instantaneously on the current value and immediately compute the
+operation. This way the user is able to build a complex expression before to
+decide to execute it and get the result. The mathematical order is respected,
+for instance `*` and `/` are computed before `+` and `-`, and parenthesis are
+available to change priorities. 
+
+To process the expression a parser is bundled with the calculator component.
+The mathematical precision is brought by an
+[arbitrary precision](https://en.wikipedia.org/wiki/Arbitrary-precision_arithmetic)
+library, in order to counter the well known
+[round-off issue](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html).
+However, some precision issues may still occur, especially when dealing with
+irrational values.
+
+## Computation engine
+The computation engine is lead by two third-party components: the parser and the
```

Comment:
I think it will be better if we say `...3 third-party components: the parser, the numbers representation engine and the tokenizer...`

Reply:
Ok, I reworded the sentence.

For the split in several files, not sure it will be better. Those sections are very small, and it will reduce the readability IMO.